tests/circuit_graph/test_circuit_graph_hash.py

```diff
@@ -56,27 +56,27 @@ def test_serialize_numeric_arguments(self, queue, observable_queue, expected_str
     observable3 = qml.prod(qml.PauliZ(0), qml.PauliZ(1))
 
     numeric_observable_queue = [
-        (returntype1, observable1, "|||ObservableReturnTypes.Expectation!PauliZ[0]"),
+        (returntype1, observable1, "|||ExpectationMP!PauliZ[0]"),
```

Comment:
QQ: after deprecating return type and replacing occurrences with isinstance, we might have to replace the expected output texts in `circuit_graph` with the exact class names. Is this a harmless change here? @albi3ro @astralcai 
Also, in this module do we have any secondary applications from these return type information?

Reply:
Another question: is `doc/development/guide/architecture.rst` also within the scope of deprecations/removals? It defines `MP.return_type` there as well

Reply:
> QQ: after deprecating return type and replacing occurrences with isinstance, we might have to replace the expected output texts in `circuit_graph` with the exact class names. Is this a harmless change here? @albi3ro @astralcai Also, in this module do we have any secondary applications from these return type information?

I see another usecase in drawer: `tape_text.py:_add_measurement` and `_add_cwire_measurement`

Reply:
Also, `return_type` has been built into the `repr` of MP...

Reply:
> Also, `return_type` has been built into the `repr` of MP...

solution: make `repr` equivalents individually for each subclass of MP.
reason: we definitely want to get rid of the Enum of obs short name list, but on the other hand we should not touch any public interfaces that use these pieces of information.

Reply:
I think we should be fine to change `CircuitGraph.serialize`.  Pretty sure it's not really used for anything anymore.

Reply:
Done at commit 5516674c9ef69cb41807ea2056a133ac2c428562